core/config/engine.cpp

```diff
@@ -210,6 +210,20 @@ void Engine::get_singletons(List<Singleton> *p_singletons) {
 	}
 }
 
+void Engine::set_buffered_logger(BufferedSingletonLogger *logger) {
+	buffered_log = logger;
+}
+
+Array Engine::get_log_buffer() {
+	ERR_FAIL_COND_V_MSG(!GLOBAL_GET("logging/buffered_logging/enable_buffered_logging"), Array(), "The project setting \"logging/buffered_logging/enable_buffered_logging\" must be `true` to get the buffered output log.");
```

Comment:
Checking project settings each time the method is called seems quite expensive, this should likely be cached so you only check a local bool (which means update needed to change the behavior).